task/move.go

```diff
@@ -25,23 +28,49 @@ type Putter func(Thing)
 // return an error.
 type MaybePutter func(Thing) error
 
+// fetch function simulates returning Thing until there's nothing else
+func fetch() (thing Thing, ok bool) {
+	bookList := []string{"concurrency with Go", "Go systems programming",
+		"Isomorphic Go", "Go BluePrints", "Master Go",
+		"Go Library Cookbook", "Algorithm and Data structures"}
+
+	// generate a random index, if the index is negative
+	// return false with an empty thing
+	rand.Seed(time.Now().UnixNano())
+	thing = rand.Intn(len(bookList)-(-len(bookList))) + (-len(bookList))
+	if thing.(int) > 0 {
+		thing, ok = bookList[thing.(int)], true
+	}
+	return
+}
+
+var storeForPut = []Thing{}
+
+// put take a value of type Thing and store it in a global map
+func put(thing Thing) {
+	storeForPut = append(storeForPut, thing)
+}
+
 // Move concurrently fetches Things from fetch() and puts them in put(). It
 // returns once fetch returns false (i.e. there are no more Things) and all
 // Things have been put().
-func Move(fetch Fetcher, put Putter) {
-	// [TODO]
-
+func Move(wg *sync.WaitGroup, fetch Fetcher, put Putter) {
```

Comment:
Keep the function signature as it was. Imagine that you're implementing something to match an interface. You pretty much never need to pass a WaitGroup around, and for this function you shouldn't need one.